src/Drupal/DrupalExtension/Context/EntityContext.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\DrupalExtension\Context;
+
+use Behat\Behat\Context\TranslatableContext;
+use Behat\Mink\Element\Element;
+
+use Behat\Gherkin\Node\TableNode;
+
+/**
+ * Provides pre-built step definitions for testing any Drupal entity.
+ */
+class EntityContext extends RawDrupalContext implements TranslatableContext
+{
+
+  /**
+   * Returns list of definition translation resources paths.
+   *
+   * @return array
+   */
+  public static function getTranslationResources()
+  {
+    return glob(__DIR__ . '/../../../../i18n/*.xliff');
+  }
+
+  /**
+   * @Given I am viewing a/an :entity_type entity/item with the :label_name :label
+   * @Given I am viewing a/an :bundle :entity_type entity/item with the :label_name :label
+   */
+  public function createEntity($entity_type, $bundle = NULL, $label_name, $label)
+  {
+    $entity = (object)array(
+        'step_bundle' => $bundle,
+        $label_name => $label,
+    );
+    $saved = $this->entityCreate($entity_type, $entity);
+    // Set internal browser on the new entity.
+    //@todo: use the entity's toUrl()
+    $this->getSession()->visit($this->locatePath('/' . $entity_type . '/' . $saved->id));
+  }
+
+  /**
+   * Creates content of a given type provided in the form:
+   * | title    | author     | status | created           |
+   * | My title | Joe Editor | 1      | 2014-10-17 8:00am |
+   * | ...      | ...        | ...    | ...               |
+   *
+   * @Given a/an :entity_type entity/item:
+   * @Given :entity_type entities/items:
+   * @Given a/an :bundle :entity_type entity/item:
+   * @Given :bundle :entity_type entities/items:
+   */
+   public function createEntities($entity_type, $bundle = NULL, TableNode $entitiesTable)
+   {
+     foreach ($entitiesTable->getHash() as $entityHash) {
+       $entity = (object)$entityHash;
+       if (!isset($entity->step_bundle)) $entity->step_bundle = $bundle;
+       $this->entityCreate($entity_type, $entity);
```

Comment:
I guess you've chosen `step_bundle` to avoid potential conflicts with actual properties like `type` and `bundle` on real entities? This is another argument for using a real class for the `$entity` instead of a plain object.